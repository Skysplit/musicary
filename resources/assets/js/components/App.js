import React, { Component } from 'react';

export class App extends Component {
  render() {
    return (
      <div>Test site</div>
    );
  }
}

export default App;
