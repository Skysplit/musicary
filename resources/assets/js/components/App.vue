<script>
const store = require('../store')

module.exports = {
    store,
    router,
    data() {
        return {
        }
    },
    created() {
        this.$auth.commit('ready')
    }
}
</script>
