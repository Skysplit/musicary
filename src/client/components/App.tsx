import React, { PureComponent } from 'react';
import ListItemLink from '@app/client/components/ListItemLink';
import { UserInterface } from '@client/store/user';
import { NavigationDrawer, ListItem } from 'react-md';
import { AppContainerProps } from '@app/client/containers/AppContainer';
import { removeUser } from '@app/client/utils/userData';

export interface AppProps extends AppContainerProps {
  user: UserInterface;
}

export default class App extends PureComponent<AppProps> {
  private logout = () => {
    removeUser();
    this.props.logout();
    this.props.router.push('/');
  }

  private guestItems() {
    return [
      <ListItem
        key="login"
        component={ListItemLink}
        route="login"
        primaryText="Login"
      />,

      <ListItem
        key="signup"
        primaryText="Sign up"
      />,
    ];
  }

  private userItems() {
    return [
      <ListItem
      key="playlists"
      primaryText="My playlists"
      component={ListItemLink}
      route="playlists"
      />,
      <ListItem
        key="logout"
        primaryText="Log out"
        onClick={this.logout}
      />,
    ];
  }

  private items() {
    if (this.props.user) {
      return this.userItems();
    }

    return this.guestItems();
  }


  render() {
    const { children, title } = this.props;


    return (
      <NavigationDrawer
        drawerHeader={false}
        toolbarTitle={title || 'Musicary'}
        navItems={[
          <ListItem
            key="home"
            component={ListItemLink}
            route="home"
            primaryText="Home"
          />,
          ...this.items(),
        ]}
      >
        {children}
      </NavigationDrawer>
    );
  }
}
