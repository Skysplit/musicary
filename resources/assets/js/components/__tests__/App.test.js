import { App } from '../App';

describe('<App />', () => {
  const defaultProps = {};
  const setup = buildSetup(App, defaultProps);

  test('renders', () => {
    expect(setup().wrapper).toMatchSnapshot();
  });
});
