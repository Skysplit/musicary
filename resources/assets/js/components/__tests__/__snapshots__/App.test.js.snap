// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <NavigationDrawer
    autoclose={true}
    constantDrawerType={false}
    contentComponent="main"
    contentId="main-content"
    defaultMedia="mobile"
    desktopDrawerType="full-height"
    desktopMinWidth={1025}
    drawerTitle="AppName"
    drawerTransitionDuration={300}
    extractMini={true}
    includeDrawerHeader={true}
    jumpLabel="Jump to content"
    mobileDrawerType="temporary"
    mobileMinWidth={320}
    persistentIconChildren="arrow_back"
    position="left"
    tabletDrawerType="temporary"
    tabletMinWidth={768}
    temporaryIconChildren="menu"
    toolbarActions={
        Array [
            <ButtonTooltipedInked
              component={[Function]}
              icon={true}
              inkTransitionEnterTimeout={450}
              inkTransitionLeaveTimeout={300}
              inkTransitionOverlap={150}
              tooltipLabel="Join AppName!"
        >
              person_add
        </ButtonTooltipedInked>,
            <ButtonTooltipedInked
              component={[Function]}
              icon={true}
              inkTransitionEnterTimeout={450}
              inkTransitionLeaveTimeout={300}
              inkTransitionOverlap={150}
              tooltipLabel="Sign in"
        >
              account_circle
        </ButtonTooltipedInked>,
          ]
    }
    toolbarThemeType="colored"
    transitionEnterTimeout={300}
    transitionName="md-cross-fade"
>
    <Route
        component={[Function]}
        exact={true}
        path="/"
    />
    <Route
        component={[Function]}
        exact={true}
        path="/login"
    />
</NavigationDrawer>,
  "nodes": Array [
    <NavigationDrawer
      autoclose={true}
      constantDrawerType={false}
      contentComponent="main"
      contentId="main-content"
      defaultMedia="mobile"
      desktopDrawerType="full-height"
      desktopMinWidth={1025}
      drawerTitle="AppName"
      drawerTransitionDuration={300}
      extractMini={true}
      includeDrawerHeader={true}
      jumpLabel="Jump to content"
      mobileDrawerType="temporary"
      mobileMinWidth={320}
      persistentIconChildren="arrow_back"
      position="left"
      tabletDrawerType="temporary"
      tabletMinWidth={768}
      temporaryIconChildren="menu"
      toolbarActions={
            Array [
                  <ButtonTooltipedInked
                    component={[Function]}
                    icon={true}
                    inkTransitionEnterTimeout={450}
                    inkTransitionLeaveTimeout={300}
                    inkTransitionOverlap={150}
                    tooltipLabel="Join AppName!"
            >
                    person_add
            </ButtonTooltipedInked>,
                  <ButtonTooltipedInked
                    component={[Function]}
                    icon={true}
                    inkTransitionEnterTimeout={450}
                    inkTransitionLeaveTimeout={300}
                    inkTransitionOverlap={150}
                    tooltipLabel="Sign in"
            >
                    account_circle
            </ButtonTooltipedInked>,
                ]
      }
      toolbarThemeType="colored"
      transitionEnterTimeout={300}
      transitionName="md-cross-fade"
>
      <Route
            component={[Function]}
            exact={true}
            path="/"
      />
      <Route
            component={[Function]}
            exact={true}
            path="/login"
      />
</NavigationDrawer>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <App />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": App {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "getDrawerOptions": [Function],
        "props": Object {},
        "refs": Object {},
        "renderLink": [Function],
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <NavigationDrawer
          autoclose={true}
          constantDrawerType={false}
          contentComponent="main"
          contentId="main-content"
          defaultMedia="mobile"
          desktopDrawerType="full-height"
          desktopMinWidth={1025}
          drawerTitle="AppName"
          drawerTransitionDuration={300}
          extractMini={true}
          includeDrawerHeader={true}
          jumpLabel="Jump to content"
          mobileDrawerType="temporary"
          mobileMinWidth={320}
          persistentIconChildren="arrow_back"
          position="left"
          tabletDrawerType="temporary"
          tabletMinWidth={768}
          temporaryIconChildren="menu"
          toolbarActions={
                    Array [
                              <ButtonTooltipedInked
                                component={[Function]}
                                icon={true}
                                inkTransitionEnterTimeout={450}
                                inkTransitionLeaveTimeout={300}
                                inkTransitionOverlap={150}
                                tooltipLabel="Join AppName!"
                    >
                                person_add
                    </ButtonTooltipedInked>,
                              <ButtonTooltipedInked
                                component={[Function]}
                                icon={true}
                                inkTransitionEnterTimeout={450}
                                inkTransitionLeaveTimeout={300}
                                inkTransitionOverlap={150}
                                tooltipLabel="Sign in"
                    >
                                account_circle
                    </ButtonTooltipedInked>,
                            ]
          }
          toolbarThemeType="colored"
          transitionEnterTimeout={300}
          transitionName="md-cross-fade"
>
          <Route
                    component={[Function]}
                    exact={true}
                    path="/"
          />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/login"
          />
</NavigationDrawer>,
        "_debugID": 2,
        "_renderedOutput": <NavigationDrawer
          autoclose={true}
          constantDrawerType={false}
          contentComponent="main"
          contentId="main-content"
          defaultMedia="mobile"
          desktopDrawerType="full-height"
          desktopMinWidth={1025}
          drawerTitle="AppName"
          drawerTransitionDuration={300}
          extractMini={true}
          includeDrawerHeader={true}
          jumpLabel="Jump to content"
          mobileDrawerType="temporary"
          mobileMinWidth={320}
          persistentIconChildren="arrow_back"
          position="left"
          tabletDrawerType="temporary"
          tabletMinWidth={768}
          temporaryIconChildren="menu"
          toolbarActions={
                    Array [
                              <ButtonTooltipedInked
                                component={[Function]}
                                icon={true}
                                inkTransitionEnterTimeout={450}
                                inkTransitionLeaveTimeout={300}
                                inkTransitionOverlap={150}
                                tooltipLabel="Join AppName!"
                    >
                                person_add
                    </ButtonTooltipedInked>,
                              <ButtonTooltipedInked
                                component={[Function]}
                                icon={true}
                                inkTransitionEnterTimeout={450}
                                inkTransitionLeaveTimeout={300}
                                inkTransitionOverlap={150}
                                tooltipLabel="Sign in"
                    >
                                account_circle
                    </ButtonTooltipedInked>,
                            ]
          }
          toolbarThemeType="colored"
          transitionEnterTimeout={300}
          transitionName="md-cross-fade"
>
          <Route
                    component={[Function]}
                    exact={true}
                    path="/"
          />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/login"
          />
</NavigationDrawer>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <App />,
}
`;
